src/mdesktopaction.h

```diff
@@ -0,0 +1,43 @@
+/***************************************************************************
+** This file was derived from the MDesktopAction implementation in the
+** libmeegotouch library.
+**
+** Copyright (C) 2021 Chupligin Sergey (NeoChapay) <neochapay@gmail.com>
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file LICENSE.LGPL included in the packaging
+** of this file.
+**
+****************************************************************************/
+
+#ifndef MDESKTOPACTION_H
+#define MDESKTOPACTION_H
+
+#include <mlite-global.h>
+#include <QString>
+
+class MDesktopEntryPrivate;
+
+class MLITESHARED_EXPORT MDesktopAction
+{
+
+public:
+    MDesktopAction(const QString &fileName, const QString &action);
+    virtual ~MDesktopAction();
+
+    QString name() const;
+    QString nameUnlocalized() const;
+    QString exec() const;
+    QString icon() const;
+    QString actionName() const;
+    QString actionEntryName() const;
```

Comment:
There's now name, nameUnlocalized, actionName and actionEntryName and that all starts to be confusing. In the end there should be only name with or without localization?

This looks pointless in the API and I'd assume actionName() should be really same as nameUnlocalized?

Reply:
Action name - "Open a New Private Window"
Action entry name - "new-private-window"

Reply:
Ah true, except in this case the "action entry name" would be "Desktop Action new-private-window".

Confusing nevertheless. Could help bit if these talked about identifiers instead of names. Though not sure how much need there is to get the identifiers. Especially this latter one with hard-coded "Desktop Action ".